File: app/src/main/java/fr/nspu/dev/recordvoicelearning/controller/dao/FolderDao.java
package fr.nspu.dev.recordvoicelearning.controller.dao;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import java.util.List;

import fr.nspu.dev.recordvoicelearning.controller.entity.FolderEntity;

/**
 * Created by nspu on 18-02-02.
 */

@Dao
public interface FolderDao {
    //sync
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertFoldersSync(List<FolderEntity> folders);
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertFolderSync(FolderEntity folder);
    @Update
    void updateFoldersSync(FolderEntity... folders);
    @Delete
    void deleteFolders(FolderEntity... folders);
    @Query("SELECT * FROM folders WHERE id = :id")
    FolderEntity loadFolderByIdSync(int id);
    @Query("SELECT * FROM folders")
    FolderEntity[] loadAllFoldersArraySync();
    @Query("SELECT * FROM folders")
    List<FolderEntity> loadAllFoldersSync();


    //async
    @Query("SELECT * FROM folders WHERE id = :id")
    LiveData<FolderEntity> loadFolderById(int id);
    @Query("SELECT * FROM folders")
    LiveData<List<FolderEntity>> loadAllFolders();
}

File: app/src/main/java/fr/nspu/dev/recordvoicelearning/controller/dao/FolderDao.kt
package fr.nspu.dev.recordvoicelearning.controller.dao

import android.arch.lifecycle.LiveData
import android.arch.persistence.room.Dao
import android.arch.persistence.room.Delete
import android.arch.persistence.room.Insert
import android.arch.persistence.room.OnConflictStrategy
import android.arch.persistence.room.Query
import android.arch.persistence.room.Update

import fr.nspu.dev.recordvoicelearning.controller.entity.FolderEntity

/**
 * Created by nspu on 18-02-02.
 */

@Dao
interface FolderDao {
    //sync
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertFoldersSync(folders: List<FolderEntity>)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertFolderSync(folder: FolderEntity)

    @Update
    fun updateFoldersSync(vararg folders: FolderEntity): Int

    @Delete
    fun deleteFolders(vararg folders: FolderEntity)

    @Query("SELECT * FROM folders WHERE id = :id")
    fun loadFolderByIdSync(id: Int): FolderEntity

    @Query("SELECT * FROM folders")
    fun loadAllFoldersArraySync(): Array<FolderEntity>

    @Query("SELECT * FROM folders")
    fun loadAllFoldersSync(): List<FolderEntity>


    //async
    @Query("SELECT * FROM folders WHERE id = :id")
    fun loadFolderById(id: Int): LiveData<FolderEntity>

    @Query("SELECT * FROM folders")
    fun loadAllFolders(): LiveData<List<FolderEntity>>
}


